aws/resource_aws_ssm_parameter.go

```diff
@@ -196,6 +204,7 @@ func resourceAwsSsmParameterPut(d *schema.ResourceData, meta interface{}) error
 		Value:          aws.String(d.Get("value").(string)),
 		Overwrite:      aws.Bool(shouldUpdateSsmParameter(d)),
 		AllowedPattern: aws.String(d.Get("allowed_pattern").(string)),
+		DataType:       aws.String(d.Get("data_type").(string)),
```

Comment:
Since the attribute has no default and it is always added to the API call here, this will trigger errors when its not configured:

```
    TestAccAWSSSMParameter_basic: testing.go:684: Step 0 error: errors during apply:

        Error: error creating SSM parameter: ValidationException: The following data type is not supported: . (Data type names are all lowercase.)
```

To handle this we should remove this line:

```suggestion
```

and instead use `d.GetOk()` right below this declaration:

```go
	if v, ok := d.GetOk("data_type"); ok {
		paramInput.DataType = aws.String(v.(string))
	}
```